<launch>
  <arg name="map_file"  default="map6"/>

  <!--  ************** Global Parameters ***************  -->
  <param name="/use_sim_time" value="false"/>

  <!--  ******************* Maps ***********************  -->
  <!-- <node name="map_server" pkg="map_server" type="map_server" args="$(find teb_local_planner_tutorials)/maps/$(arg map_file).yaml" output="screen"/> -->
    <!-- <param name="frame_id" value="map"/> -->
  <!-- </node> -->

  <!--  ************** Stage Simulator *****************  -->
  <!-- <node pkg="stage_ros" type="stageros" name="stageros" args="$(find teb_local_planner_tutorials)/stage/$(arg map_file).world">
    <remap from="/robot_0/base_scan" to="/robot_0/scan"/>
  </node> -->

  <!--  ************** Navigation ROBOT 0 *************  -->
  
    <!-- work around for footprint reload -->
  	<!-- <rosparam command="delete" ns="move_base" /> -->
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 /map /footprint 100" />
  
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/dyn_obst/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/dyn_obst/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/dyn_obst/teb_local_planner_params.yaml" command="load" />
    <!-- Here we load our costmap conversion settings -->
    <!-- If you active the following line, disable 'ground_truth_obstacles' at the bottom of this script! -->
    <!-- rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/dyn_obst/costmap_converter_params.yaml" command="load" /-->
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/diff_drive/costmap_converter_params.yaml" command="load" /> 

    <param name="TebLocalPlannerROS/include_costmap_obstacles" value="False" />
    <param name="TebLocalPlannerROS/include_dynamic_obstacles" value="True" />

    <!-- <param name="base_global_planner" value="navfn/NavfnROS" /> -->
    <param name="base_global_planner" value="global_planner/GlobalPlanner" />
    <param name="planner_frequency" value="1.0" />
    <param name="planner_patience" value="5.0" />

    <param name="base_local_planner" value="teb_local_planner/TebLocalPlannerROS" />
    <param name="controller_frequency" value="5.0" />
    <param name="controller_patience" value="15.0" />
    <!-- <remap from="map" to="/map"/> -->

    <remap from="cmd_vel" to="/mobile_base/commands/velocity"/>
    <remap from="odom" to="odom"/>
    <remap from="scan" to="scan"/>  
  </node>

  <!-- <node pkg="amcl" type="amcl" name="amcl" output="screen">
    <rosparam file="$(find teb_local_planner_tutorials)/cfg/amcl_params.yaml" command="load" />

    <param name="initial_pose_x"            value="2"/>
    <param name="initial_pose_y"            value="2"/>
    <param name="initial_pose_a"            value="0"/>
  </node> -->

  <include file="$(find teb_local_planner_tutorials)/launch/amcl.launch"/>

  

  <!--  **************** Visualisation ****************  -->
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find teb_local_planner_tutorials)/cfg/rviz_navigation_teb.rviz">
    <remap from="/move_base_simple/goal" to="/move_base_simple/goal" />
    <remap from="/initialpose" to="/initialpose"/>

    <!-- <param name="/initialpose/pose/position/x" type="double" value="0.11" /> -->
    <!-- <param name="/initialpose/pose/position/y" type="double" value="0.52" /> -->
    <!-- <remap from="/scan" to="/robot_0/scan" /> -->
  </node>


  <!-- <node name="initial_pose_publisher" pkg="rostopic" type="rostopic" args="pub -1 /initialpose geometry_msgs/PoseWithCovarianceStamped '{header: {stamp: now, frame_id: map}, pose: {pose: {position: {x: 0.11, y: 0.52, z: 0.0}, orientation: {x: 0.0, y: 0.0, z: 0.0, w: 1.0}}, covariance: [0.25, 0, 0, 0, 0, 0, 0, 0.25, 0, 0, 0, 0, 0, 0, 0.06853891945200942, 0, 0, 0, 0, 0, 0.06853891945200942, 0, 0, 0, 0, 0, 0, 0.06853891945200942]}}'"/> -->



  <!-- <node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode" output="screen">
      <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>
      <param name="serial_baudrate"     type="int"    value="115200"/>
      
      <param name="frame_id"            type="string" value="base_footprint"/>
      <param name="inverted"            type="bool"   value="false"/>
      <param name="angle_compensate"    type="bool"   value="true"/>
    </node> -->


    <include file="$(find kobuki_node)/launch/minimal.launch"/>

    
  <arg name="urdf_file" default="$(find xacro)/xacro '$(find kobuki_description)/urdf/kobuki_standalone.urdf.xacro'"/>
  <param name="robot_description" command="$(arg urdf_file)"/>
  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" type="double" value="5.0"/>
  </node>

</launch>
